src/toolkits/activity_classification/ac_data_iterator.cpp

```diff
@@ -280,6 +280,153 @@ variant_map_type _activity_classifier_prepare_data_impl(const gl_sframe &data,
+    
+    flex_vec curr_chunk_targets;
+    flex_vec curr_chunk_features;
+
+    flexible_type last_session_id = data[session_id][0];
+    size_t number_of_sessions = 0;
+
+    // Prepare an output SFrame writer, that will write a new SFrame in the converted batch-processing
+    // ready format.
+    std::vector<std::string> output_column_names = {"features", "chunk_len", "session_id"};
+    std::vector<flex_type_enum> output_column_types = {flex_type_enum::VECTOR, flex_type_enum::INTEGER,
+                                                        data[session_id].dtype()};
+    if (use_target) {
+      output_column_names.insert(output_column_names.end(), {"target"});
+      output_column_types.insert(output_column_types.end(),
+                                 {flex_type_enum::VECTOR});
+    }
+
+    gl_sframe_writer output_writer(output_column_names, output_column_types, 1);
+
+    if (verbose) {
+      logprogress_stream << "Using sequences of size " << chunk_length << " for model creation." << std::endl;
+    }
+
+    time_t last_print_time = time(0);
+    size_t processed_lines = 0;
+    int feature_size;
+
+    // Iterate over the user data. The features and targets are aggregated, and
+    // handled whenever a the ending of a session is reached.
+    int chunk_size = 0;
+    for (const auto& line: data.range_iterator()) {
+
+      auto curr_session_id = line[column_index_map[session_id]];
+
+      if (curr_session_id != last_session_id) {
+
+        feature_size = chunk_size * features.size();
+
+        // Write the aggregated data of the current chunk (which includes all
+        // the samples in the current session) as a single new vector in the
+        // converted SFrame, and init all aggregation vectors to begin a new
+        // chunk.
+        if (curr_chunk_features.size() > 0) {
+          if (use_target) {
+            output_writer.write({curr_chunk_features, chunk_size,
+                                 last_session_id, curr_chunk_targets},
+                                0);
+          } else {
+            output_writer.write(
+                {curr_chunk_features, chunk_size, last_session_id}, 0);
+          }
+          curr_chunk_features.clear();
+          curr_chunk_targets.clear();
+        }
+
+        chunk_size = 0;
+        last_session_id = curr_session_id;
+        number_of_sessions++;
+      }
+
+        chunk_size += 1;
+
+        for (const auto feature_name : features) {
+          curr_chunk_features.push_back(line[column_index_map[feature_name]]);
+        }
+
+        // If target column exists, the targets are aggregated for the duration
+        // of prediction_window.
+        // Each prediction_window subsampled into a single target value, by selecting
+        // the most frequent value (statistical mode) within the window.
+        if (use_target) {
+          curr_chunk_targets.push_back(line[column_index_map[target]]);
+        }
+
+        time_t now = time(0);
```

Comment:
I think this line should be moved into the body of the if-statement.

Reply:
I think being outside is correct. Can you look at it once more? We want print progress if its been more than 10 seconds since the last time we printed.